tests/cases.lisp

```diff
@@ -0,0 +1,66 @@
+;;;; tests/cases.lisp
+;;;;
+;;;; Runs the distributed blossom algorithm against a family of graphs with known
+;;;; MWPM, and asserts that we find the correct answer.
+
+(in-package :anatevka-tests)
+
+(eval-when (:compile-toplevel :load-toplevel :execute) ; needed at compile-time
+
+  (defun test-case-directory-filepath (dirname)
+    "Return the absolute pathname for `DIRNAME', which is the name of a test case directory in the `tests/py/cases'."
+    (declare (string dirname))
+    (let* ((anatevka-dir (ql:where-is-system "anatevka-tests")))
+      (merge-pathnames (concatenate 'string "tests/py/cases/" dirname "/*.*")
+                       anatevka-dir)))
+  
+  (defun parse-test-case-file (filename)
+    "Parses a test case file and extracts the MWPM value and the list of nodes."
+    (with-open-file (stream filename :direction :input)
+      (let ((mwpm-value (parse-integer (read-line stream))) ; read first line for mwpm
+            (nodes '()))
+        (loop :for line := (read-line stream nil)
+              :while line
+              :do (destructuring-bind (x y)
+                     (mapcar #'parse-integer (uiop:split-string line :separator ","))
+                   (push (list x y) nodes)))
+        (values mwpm-value (reverse nodes)))))
+
+  (defun process-test-case-directory (dirname)
+    "Iterates through all files in test case directory named `DIRNAME' and parses them."
+    (let* ((dirpath (test-case-directory-filepath dirname))
+           (files (directory dirpath)))
+      (loop :for file :in files
+            :collect (list file (multiple-value-list (parse-test-case-file file)))
+              :into test-cases
+            :finally (return test-cases))))
+
+  (defun sanitize-test-name (dirname test-path)
+    "Generates a valid Lisp symbol for a test name based on directory and file name."
+    (let* ((test-name (pathname-name test-path))
+           (clean-name (format nil "test-blossom-suite-~A-~A" dirname test-name)))
+      (intern (string-upcase (substitute #\- #\/ clean-name)) :anatevka-tests))))
```

Comment:
Honestly this seems fine, since there aren't a bajillion test cases, but I have a dull memory that it's ill-advised to intern programmatically generated symbols. As an alternative, I guess we could just iterate through the directory at eval time of a single umbrella test, whether than at compile time to generate individuated tests?

Don't worry about it, just a comment.